java/query/query-lambda/src/test/java/sleeper/query/lambda/WarmQueryExecutorLambdaTest.java

```diff
@@ -0,0 +1,108 @@
+import sleeper.core.schema.type.ByteArrayType;
+import sleeper.core.schema.type.IntType;
+import sleeper.core.schema.type.LongType;
+import sleeper.core.schema.type.StringType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class WarmQueryExecutorLambdaTest {
+
+    @Test
+    public void testStringTypeRegionIsReturned() {
+        Schema schema = getStringKeySchema();
+        Region region = WarmQueryExecutorLambda.getRegion(schema);
+
+        assertThat(region).isEqualTo(expectedRegion(schema, "a"));
+    }
+
+    @Test
+    public void testByteTypeRegionIsReturned() {
+        Schema schema = getByteArrayKeySchema();
+        Region region = WarmQueryExecutorLambda.getRegion(schema);
+
+        assertThat(region).isEqualTo(expectedRegion(schema, new byte[]{'a'}));
+    }
+
+    @Test
+    public void testIntTypeRegionIsReturned() {
+        Schema schema = getIntKeySchema();
+        Region region = WarmQueryExecutorLambda.getRegion(schema);
+
+        assertThat(region).isEqualTo(expectedRegion(schema, 0));
+    }
+
+    @Test
+    public void testLongTypeRegionIsReturned() {
+        Schema schema = getLongKeySchema();
+        Region region = WarmQueryExecutorLambda.getRegion(schema);
+
+        assertThat(region).isEqualTo(expectedRegion(schema, 0L));
+    }
+
+    private Schema getStringKeySchema() {
+        return Schema.builder()
+                .rowKeyFields(new Field("test-key", new StringType()))
+                .sortKeyFields(new Field("test-sort", new StringType()))
+                .valueFields(new Field("test-value", new StringType()))
+                .build();
+    }
+
+    private Schema getByteArrayKeySchema() {
+        return Schema.builder()
+                .rowKeyFields(new Field("test-key", new ByteArrayType()))
+                .sortKeyFields(new Field("test-sort", new ByteArrayType()))
+                .valueFields(new Field("test-value", new ByteArrayType()))
+                .build();
+    }
+
+    private Schema getIntKeySchema() {
+        return Schema.builder()
+                .rowKeyFields(new Field("test-key", new IntType()))
+                .sortKeyFields(new Field("test-sort", new IntType()))
+                .valueFields(new Field("test-value", new IntType()))
+                .build();
+    }
+
+    private Schema getLongKeySchema() {
+        return Schema.builder()
+                .rowKeyFields(new Field("test-key", new LongType()))
+                .sortKeyFields(new Field("test-sort", new LongType()))
+                .valueFields(new Field("test-value", new LongType()))
+                .build();
+    }
+
+    private Region expectedRegion(Schema schema, Object value) {
+        List<Range> ranges = new ArrayList<>();
+        schema.getRowKeyFields().forEach(field -> {
+            ranges.add(new Range.RangeFactory(schema)
+                    .createExactRange(field, value));
+        });
```

Comment:
This computes which fields should have ranges. In order to tell what the tests are asserting, you need to read this, and figure out whether it's correct in the same way you would figure out whether the production code is correct. It's pretty easy to imagine changing the production code in a way which would produce incorrect behaviour, then changing this code to match it without working out whether the tests are asserting what we want.

We can avoid that if we construct the data we want to assert on directly. We can build ranges specifically for the fields we expect there to be ranges for. We want to see the data directly in the test so we can see whether it describes the scenario we expect.